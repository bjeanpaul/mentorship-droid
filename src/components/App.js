import React from 'react-native';
import { Provider } from 'react-redux';

import configureStore from '../stores/configureStore';
const store = configureStore();


console.log('----------------------------')
import { login } from '../mentor/MentorActions';
store.dispatch(login('admin', '123'));
console.log('----------------------------')


import RootNavigator from './RootNavigator';



const App = function App() {
  return (
    <Provider store={store}>
        <RootNavigator />
    </Provider>
  );
};

export default App;
