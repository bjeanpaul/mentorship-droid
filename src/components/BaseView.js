import React from 'react';
import { View, StyleSheet, StatusBar } from 'react-native';
import colors from 'src/constants/colors';


const styles = StyleSheet.create({
  view: {
    backgroundColor: colors.DEFAULT_BG,
    flex: 1,
  },
});

const BaseView = ({
  children,
  style,
  statusBarBackgroundColor = colors.DEFAULT_STATUS_BG,
}) => (
  <View style={[styles.view, style]}>
    { statusBarBackgroundColor
      ? <StatusBar backgroundColor={statusBarBackgroundColor} />
      : null
    }
    {children}
  </View>
);
BaseView.propTypes = {
  statusBarBackgroundColor: React.PropTypes.string,
  children: React.PropTypes.any,
  style: React.PropTypes.any,
};
export default BaseView;
