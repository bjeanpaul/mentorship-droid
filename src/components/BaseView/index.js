import React from 'react';
import { View } from 'react-native';

import styles from './styles';


const BaseView = ({
  children,
  style,
}) => (
  <View style={[styles.view, style]}>
    {children}
  </View>
);


BaseView.propTypes = {
  children: React.PropTypes.any,
  style: React.PropTypes.any,
};


export default BaseView;
