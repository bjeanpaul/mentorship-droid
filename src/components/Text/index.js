import React, { PropTypes } from 'react';
import { Text as RNText } from 'react-native';
import styles, { types, themes, STYLE_TRANSFORMS } from './styles';
import transform from './transform';


const Text = ({
  children,
  style,
  ...props,
}) => {
  const text = transform(children, style, STYLE_TRANSFORMS);

  return (
    <RNText
      {...props}
      style={[styles.default, style]}
    >
      {text}
    </RNText>
  );
};


Text.propTypes = {
  children: PropTypes.node,
  style: PropTypes.any,
};


Text.types = types;
Text.themes = themes;


export default Text;
