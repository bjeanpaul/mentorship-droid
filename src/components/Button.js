import React, { PropTypes } from 'react';
import { TouchableNativeFeedback, View, StyleSheet } from 'react-native';

import Text from 'Text';
import { font, fontWeight } from 'src/app/styles';


const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#f77040',
    borderRadius: 5,
    marginLeft: 16,
    marginRight: 16,
  },
  buttonDisabled: {
    backgroundColor: '#dfe5e6',
  },
  label: {
    color: '#fff',
    fontFamily: font.brandonTextBold,
    fontSize: 13,
    fontWeight: fontWeight.bold,
    letterSpacing: 1,
    marginTop: 15,
    marginBottom: 17,
  },
});

const Button = ({
  label,
  disabled,
  handlePress,
}) => {
  let button = (
    <View style={[styles.button, disabled && styles.buttonDisabled]}>
      <Text style={styles.label}>{label.toUpperCase()}</Text>
    </View>
  );
  if (!disabled) {
    button = (
      <TouchableNativeFeedback onPress={handlePress}>
        {button}
      </TouchableNativeFeedback>
    );
  }
  return button;
};
Button.propTypes = {
  label: PropTypes.string.isRequired,
  handlePress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};
export default Button;
