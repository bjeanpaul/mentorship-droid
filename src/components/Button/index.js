import React, { PropTypes } from 'react';
import { TouchableNativeFeedback, View } from 'react-native';
import Text from 'src/components/Text';
import styles, { themes, layouts } from './styles';


const Button = ({
  theme = themes.default,
  layout = layouts.default,
  children,
  disabled,
  onPress,
}) => {
  let button = (
    <View
      style={[
        styles.container,
        theme.container,
        layout,
        disabled && theme.containerIsDisabled,
      ]}
    >
      <Text
        style={[
          styles.title,
          theme.title,
          disabled && theme.titleIsDisabled,
        ]}
      >
        {children}
      </Text>
    </View>
  );
  if (!disabled) {
    button = (
      <TouchableNativeFeedback onPress={onPress}>
        {button}
      </TouchableNativeFeedback>
    );
  }
  return button;
};


Button.themes = themes;
Button.layouts = layouts;


Button.propTypes = {
  children: PropTypes.any,
  onPress: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  theme: PropTypes.any,
  layout: PropTypes.any,
};


export default Button;
