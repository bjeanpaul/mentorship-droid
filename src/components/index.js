import Text from './Text';
import Label from './Label';
import Link from './Link';
import Button from './Button';
import TextInput from './TextInput';
import MultiLineTextInput from './MultiLineTextInput';
import BaseView from './BaseView';
import Toolbar from './Toolbar';
import Header from './Header';
import Title from './Title';
import Icon from './Icon';
import HeaderIcon from './HeaderIcon';
import OverlayCompleted from './Overlay/Completed';
import OverlayLoading from './Overlay/Loading';
import Section from './Section';

export {
  BaseView,
  Toolbar,
  Text,
  Label,
  Link,
  Button,
  TextInput,
  MultiLineTextInput,
  Toolbar,
  Header,
  Title,
  Icon,
  HeaderIcon,
  OverlayCompleted,
  OverlayLoading,
  Section,
};
