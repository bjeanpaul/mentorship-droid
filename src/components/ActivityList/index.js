import React, { PropTypes } from 'react';
import { View, Image, ScrollView, TouchableNativeFeedback } from 'react-native';
import { Text } from 'src/components';
import images from 'src/constants/images';
import { ImageUrl } from 'src/api';
import styles from './styles';


const ActivityList = ({
  category: { color },
  activities,
  onActivityPress,
}) => (
  <ScrollView>
    {activities.map(({ id, title, icon, isComplete }) => (
    <TouchableNativeFeedback
      key={id}
      uid={`activity:${id}`}
      onPress={() => onActivityPress(id)}
    >
      <View style={styles.activity}>
        <ActivityIcon icon={icon} color={color} isComplete={isComplete} />
        <Text numberOfLines={3} style={styles.activityTitle}>{title}</Text>
      </View>
    </TouchableNativeFeedback>
    ))}
  </ScrollView>
);


const ActivityIcon = ({
  icon,
  color,
  isComplete,
}) => (
  <View style={styles.activityIconContainer}>
    <View
      style={[
        styles.activityIcon,
        isComplete
          ? { backgroundColor: color }
          : styles.activityIconIsIncomplete,
      ]}
    >
      <Image
        source={icon.toSource(images.ACTIVITY_ICON_FALLBACK)}
        style={styles.activityIconImage}
      />
    </View>

    {
      isComplete
        ? <Image
          style={styles.activityCompleteAnnotation}
          source={images.ACTIVITY_COMPLETE_ANNOTATION}
        />
        : null
    }
  </View>
);


ActivityList.propTypes = {
  category: PropTypes.object.isRequired,
  activities: PropTypes.array.isRequired,
  onActivityPress: PropTypes.func.isRequired,
};


ActivityIcon.propTypes = {
  icon: PropTypes.instanceOf(ImageUrl),
  color: PropTypes.string.isRequired,
  isComplete: PropTypes.bool.isRequired,
};


export default ActivityList;
