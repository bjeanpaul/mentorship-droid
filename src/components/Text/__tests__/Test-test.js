import React from 'react';
import { View } from 'react-native';
import { Text } from 'src/components';


describe('Text', () => {
  it('should render', () => {
    expect(render(
      <Text>
        This is normal text. Lorem ipsum dolor sit amet, consectetur adipiscing
        elit, sed do eiusmod tempor incididuntut labore et dolore magna aliqua.
        Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi
        ut aliquip ex ea commodo consequat.
      </Text>
    )).toMatchSnapshot();
  });

  it('should render titles', () => {
    expect(render(
      <Text style={Text.types.title}>
        This is a section title. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididuntut labore et dolore
        magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation
        ullamco laboris nisi ut aliquip ex ea commodo consequat.
      </Text>
    )).toMatchSnapshot();
  });

  it('should render secondary titles', () => {
    expect(render(
      <Text style={Text.types.secondaryTitle}>
        This is a section title. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididuntut labore et dolore
        magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation
        ullamco laboris nisi ut aliquip ex ea commodo consequat.
      </Text>
    )).toMatchSnapshot();
  });

  it('should render paragraphs', () => {
    expect(render(
      <Text style={Text.types.paragraph}>
        This is a paragraph. Lorem ipsum dolor sit amet, consectetur adipiscing
        elit, sed do eiusmod tempor incididuntut labore et dolore magna aliqua.
        Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi
        ut aliquip ex ea commodo consequat.
      </Text>
    )).toMatchSnapshot();
  });

  it('should render section titles', () => {
    expect(render(
      <Text style={Text.types.sectionTitle}>
        This is a section title. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit, sed do eiusmod tempor incididuntut labore et dolore
        magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation
        ullamco laboris nisi ut aliquip ex ea commodo consequat.
      </Text>
    )).toMatchSnapshot();
  });

  it('should support a light theme', () => {
    expect(render(
      <View style={{ backgroundColor: '#000' }}>
        <Text style={Text.themes.light}>Title with theme</Text>
      </View>
    )).toMatchSnapshot();
  });
});
