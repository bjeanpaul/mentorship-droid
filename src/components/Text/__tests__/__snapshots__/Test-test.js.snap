exports[`Text should render 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#003035",
        "fontFamily": "Brandon Text",
        "fontSize": 13,
        "textAlign": "center"
      },
      undefined
    ]
  }>
  This is normal text. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididuntut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
</Text>
`;

exports[`Text should render paragraphs 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#003035",
        "fontFamily": "Brandon Text",
        "fontSize": 13,
        "textAlign": "center"
      },
      Object {
        "fontSize": 18,
        "paddingBottom": 12,
        "paddingTop": 18,
        "textAlign": "left"
      }
    ]
  }>
  This is a paragraph. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididuntut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
</Text>
`;

exports[`Text should render section titles 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#003035",
        "fontFamily": "Brandon Text",
        "fontSize": 13,
        "textAlign": "center"
      },
      Object {
        "color": "#9fb1b3",
        "fontFamily": "Brandon Text_bold",
        "fontSize": 12,
        "fontWeight": "bold",
        "letterSpacing": 1,
        "textAlign": "left"
      }
    ]
  }>
  THIS IS A SECTION TITLE. LOREM IPSUM DOLOR SIT AMET, CONSECTETUR ADIPISCING ELIT, SED DO EIUSMOD TEMPOR INCIDIDUNTUT LABORE ET DOLORE MAGNA ALIQUA. UT ENIM AD MINIM VENIAM, QUIS NOSTRUD EXERCITATION ULLAMCO LABORIS NISI UT ALIQUIP EX EA COMMODO CONSEQUAT.
</Text>
`;

exports[`Text should render titles 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#003035",
        "fontFamily": "Brandon Text",
        "fontSize": 13,
        "textAlign": "center"
      },
      Object {
        "fontFamily": "Brandon Text_medium",
        "fontSize": 20,
        "fontWeight": "400"
      }
    ]
  }>
  This is a section title. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididuntut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
</Text>
`;

exports[`Text should support a light theme 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000"
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#003035",
          "fontFamily": "Brandon Text",
          "fontSize": 13,
          "textAlign": "center"
        },
        Object {
          "color": "#fff"
        }
      ]
    }>
    Title with theme
  </Text>
</View>
`;
